### PR TITLE
MODPUBSUB-43 Created Publishing Service

### PR DESCRIPTION
**Pull request by KaterynaSenchenko**
AuditMessageService will be remade as Vertx service proxy in scope of https://issues.folio.org/browse/MODPUBSUB-49

### CODE REVIEW

File: mod-pubsub-server/src/main/java/org/folio/services/impl/PublishingServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import io.vertx.kafka.client.producer.KafkaProducer;
+import io.vertx.kafka.client.producer.impl.KafkaProducerRecordImpl;
+import org.folio.kafka.PubSubConsumerConfig;
+import org.folio.rest.jaxrs.model.Event;
+import org.folio.services.PublishingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PublishingServiceImpl implements PublishingService {
```
[the remark this answers is not in the document]
Comment (Igor-Gorchakov) [reply mod-pubsub-server/src/main/java/org/folio/services/impl/PublishingServiceImpl.java]: The implementation is lightly coupled to the KafkaProducer, let's rename it from 'PublishingServiceImpl' to 'KafkaPublisherService'

File: mod-pubsub-server/src/main/java/org/folio/services/impl/PublishingServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+import java.util.Date;
+import java.util.UUID;
+
+import static java.lang.String.format;
+import static org.folio.rest.jaxrs.model.MessagingModule.ModuleRole.PUBLISHER;
+import static org.folio.rest.jaxrs.model.MessagingModule.ModuleRole.SUBSCRIBER;
+
+@Component
+public class PublishingServiceImpl implements PublishingService {
```
Comment (Igor-Gorchakov) [inline mod-pubsub-server/src/main/java/org/folio/services/impl/PublishingServiceImpl.java]: The implementation is tightly coupled to the KafkaProducer, let's rename it from 'PublishingServiceImpl' to 'KafkaPublisherService'

Comment (OleksiiKuzminov) [reply mod-pubsub-server/src/main/java/org/folio/services/impl/PublishingServiceImpl.java]: I agree. We can use "interface PublishingService" with several implementations. And KafkaPublisherServiceImpl seems like a good naming for it

Comment (KaterynaSenchenko) [reply mod-pubsub-server/src/main/java/org/folio/services/impl/PublishingServiceImpl.java]: done, thank you

**Pull request merged**